### ISSUE TITLE
[validator] error on 'start'

### ISSUE THREAD

**Issue by iranzo** [nymtech/nym#596]
During validator start (new validator), I'm getting this:

![image](https://user-images.githubusercontent.com/312463/117423921-854b0580-af21-11eb-93e7-00934b2fe96f.png)

~~~
10:45AM INF starting ABCI with Tendermint                                                                                                                                                                          
10:45AM INF Starting multiAppConn service impl=multiAppConn module=proxy                                                                                                                                           
10:45AM INF Starting localClient service connection=query impl=localClient module=abci-client                                                                                                                      
10:45AM INF Starting localClient service connection=snapshot impl=localClient module=abci-client                                                                                                                   
10:45AM INF Starting localClient service connection=mempool impl=localClient module=abci-client                                                                                                                    
10:45AM INF Starting localClient service connection=consensus impl=localClient module=abci-client
10:45AM INF Starting EventBus service impl=EventBus module=events                   
10:45AM INF Starting PubSub service impl=PubSub module=pubsub                                  
10:45AM INF Starting IndexerService service impl=IndexerService module=txindex                  
10:45AM INF ABCI Handshake App Info hash=",\bZ���@�Y�����]�{p��ִ_�=}����B�" height=11455 module=consensus protocol-version=0 software-version=
10:45AM INF ABCI Replay Blocks appHeight=11455 module=consensus stateHeight=11455 storeHeight=11456                                                                                                                
10:45AM INF Replay last block using real app module=consensus                                                                                                                                                      
Error: error during handshake: error on replay: wrong Block.Header.AppHash.  Expected 2C085A9CB4AD40F759CBF6A1A5F45DAF7B7087B5D6B45FAA3D7DB1C1E1DD42F3, got 3F89CEA021DB28CDC6DE5CB4CAB4B87DB18DAD305518B1E89C285CB
14024853F 
~~~

**Comment by iranzo** [comment]
Note: compiled from wasmd repository and performing manual steps in 'develop' branch of docs

**Comment by futurechimp** [comment]
Hi @iranzo, we haven't seen that before. Can you have a look at some of the suggested solutions in https://github.com/binance-chain/node-binary/issues/87 and see if they work for you? That's the most similar thing I can see.

**Comment by iranzo** [comment]
Thanks @futurechimp tried the settings described at https://github.com/binance-chain/node-binary/issues/87#issuecomment-582043991 (plus removal of the data and the other file) and it came to this status:

![image](https://user-images.githubusercontent.com/312463/117474361-230ef680-af5b-11eb-8560-b5dd7bfebe5f.png)

With this traceback:

![image](https://user-images.githubusercontent.com/312463/117474494-3f129800-af5b-11eb-8ba8-f5a7d79f3bff.png)

**Comment by futurechimp** [comment]
Hm, I'll have to take a look in the next few days (I'm traveling and am not set up to have a closer look just now). 

In the meantime:

* Just to check, what git tag do you have checked out?
* What OS are you on?

If you're feeling sporty, try blowing away all the config files in `~/.nymd` and reinitializing to see what happens. 

Sorry for the hassle and we will get to the bottom of it.

**Comment by iranzo** [comment]
No worries, enjoy the trip.

What I'm doing right now is to start over again after removing the 'data' and the other file *again* but instead of my compiled version out of wasmd 'master' branch, use the one in the repo that came out for the nym current 'mixnode' release (using the tag).

I'm running this on Fedora Server 34.

As the initial load seems to be still running I'll update this once it finishes before doing the 'tx' part of the validator.

Thanks!

**Comment by iranzo** [comment]
Hi,
I' think that it's already passed over (it's still running) so I've tried the next step in the guide as I was unsure about if it should 'stop' or keep syncing(which I guess it's the thing to do).

`[root@bots build]# ./nymd tx staking create-validator   --amount=10000000stake   --fees=5000uhal   --pubkey="$PUB_KEY"   --moniker=$MONIKER   --chain-id=testnet-finney   --commission-rate="0.10"   --commission-max-rate="0.20"   --commission-max-change-rate="0.01"   --min-self-delegation="1"   --gas="auto"   --gas-adjustment=1.15   --from=$FROM_ACCOUNT   --node https://testnet-finney-validator.nymtech.net:443
Error: The specified item could not be found in the keyring
`

Seems that I cannot register it, so I was also unsure about if the 'start' process should end or now I'm on a different stage.

BTW; I got to actual status, as stated in previous comment by using the precompiled version instead of following the 'wasmd' compilation on my side.

If my current status is that it's ok, and the 'tx' needs to happen to get a valid validator I wouldn't touch this, but I could reproduce the issue with compilation as I used another machine to test and got to the same status using the "compiled-by-me" version, so we could use that one as reproducer for further debugging
